pkg/kubelego/watch.go

```diff
@@ -49,10 +61,44 @@ func (kl *KubeLego) WatchReconfigure() {
 			if quit {
 				return
 			}
-			kl.Log().Debugf("worker: begin processing %v", item)
-			kl.Reconfigure()
-			kl.Log().Debugf("worker: done processing %v", item)
-			kl.workQueue.Done(item)
+			func(item interface{}) {
+				defer kl.workQueue.Done(item)
```

Comment:
Yep - they serve two different purposes:

- *Done* will inform the queue that the particular work item has finished processing, as the workqueue will stop the same item being processed by two workers at once (although we actually only process one certificate at once anyway).

- *Forget* will forget the item altogether and clear its back off status.

So 'Done' should be called after after the function processing the results of Get() is done (regardless of err/success)

'Forget' should be called once the processing has 'succeeded' and we want to clear the rate limit state for that particular item.

Calling AddRateLimit along with Done will therefore *not* clear the rate limit. It will mark the queue as finished processing that particular item, and also queue the item to be re-added to the queue after the rate limit is up.

Reply:
Looks like you have to call both:
```
	// Forget indicates that an item is finished being retried.  Doesn't matter whether its for perm failing
	// or for success, we'll stop the rate limiter from tracking it.  This only clears the `rateLimiter`, you
	// still have to call `Done` on the queue.
	Forget(item interface{})
```